### ISSUE TITLE
Attempts to update settings.php fail on install 

### ISSUE THREAD

**Issue by dustinleblanc** [pantheon-systems/drops-8#36]
Getting this trying to install Drupal 8 with beta10
```
Warning: file_put_contents(/srv/bindings/c4d5b76556834511adc9eabdd3fc78b7/code/sites/default/settings.php): failed to open stream: Permission denied in drupal_rewrite_settings() (line 319 of core/includes/install.inc).
drupal_rewrite_settings(Array)
install_write_profile(Array)
install_run_task(Array, Array)
install_run_tasks(Array)
install_drupal(Object)
Exception: Failed to modify <em class="placeholder">sites/default/settings.php</em>. Verify the file permissions. in drupal_rewrite_settings() (line 320 of core/includes/install.inc).
drupal_rewrite_settings(Array)
install_write_profile(Array)
install_run_task(Array, Array)
install_run_tasks(Array)
install_drupal(Object)
```

Install stops there

**Comment by dustinleblanc** [comment]
Running my installer locally and commiting the changes, then pushing up to Pantheon solved the problem.

The diff was just this line:
`$settings['install_profile'] = 'standard';` so it seems adding a install profile line to the settings file would alleviate the issue

**Comment by populist** [comment]
This is a known issue since Drupal 8 requires write access to the settings.php to write this information. Our default is SFTP mode which allows this writability ,but presumably you are in Git mode. Leaving this open until we figure out a way to get this value set in both SFTP + Git mode.

**Comment by weitzman** [comment]
I wrote a patch which will avoid the problem for all sites based on Distros - https://www.drupal.org/node/2156401. Please help review.

**Comment by seanr** [comment]
Yep, I had to switch back to SFTP and was then able to proceed with the install.  Perhaps you guys can set the permissions of that directory differently for install?  Or at least some help text around this - I figured it out on my own pretty quickly, but less experienced users (or even long-term Drupal folks without Pantheon experience) might not. :-)

**Comment by populist** [comment]
It would be ideal to get Moshe's patch to work in Drupal so we can install in either case, otherwise we can just provide a bit more documentation to tell users that first time installs need to happen in settings.php.

**Comment by weitzman** [comment]
I could use help with that issue. I ran out of steam and dont plan to return to it.

**Comment by greg-1-anderson** [comment]
The big bug-a-boo here is the [question catch raised](https://www.drupal.org/node/2156401#comment-10196519).  If people are really concerned that someone might run into problems by converting a single-distro multisite into a multiple-distro multisite, then the issue will probably never get committed.  No one responded to that concern, though; maybe this means the concern is minor enough that it might move forward anyway?

At this point, we can install cleanly without it, so the benefit we get is that it would allow us to install in 'git' mode, instead of requiring a switch to sftp mode.  I don't think that's critical for us, but I might return there as time permits if it seemed the efforts might be fruitful.

**Comment by 4aficiona2** [comment]
Looks like this is still an issue. Tried to install a site on Pantheon in git mode and get the following error:

```
Warning: file_put_contents(sites/default/settings.php): failed to open stream: Permission denied in drupal_rewrite_settings() (line 337 of core/includes/install.inc).
drupal_rewrite_settings(Array)
install_write_profile(Array)
install_run_task(Array, Array)
install_run_tasks(Array)
install_drupal(Object)
``` 

Because the issue got closed I assumed that it got also resolved. What is the current state?

**Comment by greg-1-anderson** [comment]
This issue is open.  See also the issue on Drupal.org: https://www.drupal.org/node/2156401. If that is fixed and committed to Drupal core, then installation in "git" mode will work on Pantheon.  For now, though, you must switch to SFTP mode before installing.

**Comment by heddn** [comment]
Where's the status on this and on relocatable docroot?

**Comment by greg-1-anderson** [comment]
The main problem with this issue is that even once https://www.drupal.org/node/2156401 is committed, it doesn't really help the DX here unless we (a) make a 'Pantheon' distribution in drops-8, or (b) write the 'standard' profile into settings.php.  Either of those options would make it inconvenient for folks who want to install the 'minimal' profile. Because of that, we probably won't take any action vis-a-vis 2156401 in drops-8 -- we will inherit this fix once it is committed to Drupal core. Therefore, installing in git mode will only be available for those who update their settings.php prior to installing.  Practically speaking, this will probably mostly be folks who set up a custom upstream.

**Comment by deborah-ufw** [comment]
I've just run into this problem and have not been able to re-run the installer after switching to SFTP in my Pantheon dashboard. 5 minutes into Pantheon and the install failure is compromising - an error as basic as getting Drupal installed creates a poor start for evaluating Pantheon.

Using an SFTP tool (Filezilla) to try to manually update settings.php also did not work for me. Neither did pushing my local install (which succeeded).

**Comment by ari-gold** [comment]
@deborah-ufw Can you try again with another site? I was just able to install successfully. If you're still having issues, please contact support. 

The fix is upstream, not something directly in our control, but since this issue was opened, we added an error message to prompt you to change to Git mode for installation:
![screenshot 2016-11-21 11 23 32](https://cloud.githubusercontent.com/assets/7874556/20497334/a8520c92-afdd-11e6-80fd-b3601bd4d4dc.png)

Successful install today:
![screenshot 2016-11-21 11 28 56](https://cloud.githubusercontent.com/assets/7874556/20497360/b63b1efc-afdd-11e6-92bc-967f4d655a15.png)

**Comment by greg-1-anderson** [comment]
I was also able to install successfully using the minimal installation profile.

**Comment by GuyPaddock** [comment]
Looks like SFTP mode is still required for installs:
![image](https://user-images.githubusercontent.com/2631799/88815211-1322e480-d189-11ea-8f61-e76da200affe.png)

**Comment by dustinleblanc** [comment]
Correct, but now that its been that way long enough to graduate college, I've just learned to live with it :)

**Comment by weitzman** [comment]
FWIW, my patch was accepted into Drupal core 3 years ago so it may be possible for Pantheon to get past the limitation.

And yeah, perhaps its no longer a priority for anyone, given the duration of the limitation.

**Comment by deborah-ufw** [comment]
Thank you for the patch !! I was amazed to get a notification 5 years after commenting on this thread.

> And yeah, perhaps its no longer a priority for anyone, given the duration of the limitation.

Pantheon may have difficulty evaluating how many potential users have dropped off (and will continue to drop off) when they experience this kind of a failure on an install where the purpose is to evaluate the platform. Blocking first time users is usually a business killer.

**Comment by dustinleblanc** [comment]
I'd agree, but the platform now warns you specifically about what the cause of the issue is, and I believe it even spins up Drupal 8 sites in SFTP mode, so the experience is not the same as it was when us fledgling D8 developers tried to use the new fancy and hit problems. I think that is why the issue went four years without a comment :)

**Comment by powpow12** [comment]
Confirmed that I got the notification today while trying to install on git. Just my two cents, I was concerned that switching from git to sftp was going to break something. I didn't feel confident it wouldn't until I found this issue and read the comments.

**Comment by namespacebrian** [comment]
A fix has been merged so it's no longer necessary to be in "SFTP mode" to install Drupal.  The configuration change will roll out gradually to existing sites.. new sites should get it immediately.

I'll leave this issue open for a couple days and close it if no issues arise.
